stone-java-gradle-plugin/src/main/kotlin/com/dropbox/stone/java/StonePlugin.kt

```diff
@@ -37,50 +47,43 @@ public class StonePlugin : Plugin<Project> {
         } else {
             "generate${sourceSet.name.capitalize()}Stone"
         }
-        val stoneTaskProvider: TaskProvider<StoneTask> = project.tasks.register(taskName, StoneTask::class.java)
-        stoneTaskProvider.get().also { stoneTask ->
-            stoneTask.description = "Generate Stone Java source files for ${sourceSet.name}."
+
+        project.tasks.register(taskName, StoneTask::class.java){
+            description = "Generate Stone Java source files for ${sourceSet.name}."
 
             val routeWhitelistFilterPropName = "com.dropbox.api.${sourceSet.name}.routeWhitelistFilter"
             val routeWhitelistFilterValue: String? = project.properties[routeWhitelistFilterPropName] as String?
             routeWhitelistFilterValue?.let {
-                stoneTask.routeWhitelistFilter(it)
+                routeWhitelistFilter(it)
             }
 
             val specDirPropName = "com.dropbox.api.${sourceSet.name}.specDir"
             val specDirPropNameValue: String? = project.properties[specDirPropName] as String?
             val mySpecDir: String = specDirPropNameValue ?: "src/${sourceSet.name}/stone"
 
-            stoneTask.specDir(mySpecDir)
-            stoneTask.generatorDir("${project.projectDir.absoluteFile}/generator")
-            stoneTask.stoneDir("stone")
-            stoneTask.pythonCommand("python")
-            stoneTask.outputDir("${project.buildDir}/generated/source/stone/${sourceSet.name}")
-            stoneTask.inputs
-                .dir { project.fileTree(stoneTask.getGeneratorDir()) { filter { !it.name.endsWith(".pyc") } } }
+            specDir(mySpecDir)
+            generatorDir("${project.projectDir.absoluteFile}/generator")
+            stoneDir("stone")
+            pythonCommand("python")
+            outputDir("${project.buildDir}/generated/source/stone/${sourceSet.name}")
+            inputs
+                .dir { project.fileTree(this.getGeneratorDir()) { filter { !it.name.endsWith(".pyc") } } }
                 .withPropertyName("stone")
                 .withPathSensitivity(PathSensitivity.RELATIVE)
 
-            val specDir = stoneTask.getSpecDir().get()
-            stoneTask.inputs
-                .dir(stoneTask.getSpecFiles(project.objects, specDir))
+            val specDir = this.getSpecDir().get()
+            inputs
+                .dir(this.getSpecFiles(project.objects, specDir))
                 .withPathSensitivity(PathSensitivity.RELATIVE)
                 .withPropertyName("stoneSpec")
                 .skipWhenEmpty(true)
-            stoneTask.outputs
-                .dir { stoneTask.getOutputDir() }
+            outputs
+                .dir { this.getOutputDir() }
                 .withPropertyName("generatedStone")
-            stoneTask.outputs.cacheIf { true }
+            outputs.cacheIf { true }
 
-            val compileJava: Task = project.tasks.getByName(sourceSet.getCompileTaskName("java"))
```

Comment:
These dependency declarations were moved to the top of this file and are applied after the stoneTasks are created.